test/integration/edition_edit_test.rb

```diff
@@ -225,10 +225,86 @@ class EditionEditTest < IntegrationTest
     end
   end
 
+  context "edit tab" do
+    setup do
+      visit_edition_in_draft
+      click_link("Edit")
+    end
+
+    should "show 'Metadata' header and an update button" do
+      within :css, "h2.gem-c-heading" do
+        assert page.has_text?("Edit")
+      end
+      assert page.has_button?("Save")
+    end
+
+    should "show Title input box prefilled" do
+      assert page.has_text?("Title")
+      assert page.has_field?("edition[title]", with: "Edit page title")
+    end
+
+    should "show Meta tag input box prefilled" do
+      assert page.has_text?("Meta tag description")
+      assert page.has_text?("Some search engines will display this if they cannot find what they need in the main text")
+      assert page.has_field?("edition[overview]", with: "metatags")
+    end
+
+    should "show Beta content radios prechecked" do
+      assert page.has_text?("Is this beta content?")
+      assert find(".gem-c-radio input[value='0']")
+      assert find(".gem-c-radio input[value='1']").checked?
+    end
+
+    should "show Body text field prefilled" do
+      assert page.has_text?("Body")
+      assert page.has_text?("Refer to the Govspeak guidance (opens in new tab)")
+      assert page.has_field?("edition[body]", with: "The body")
+    end
+
+    should "not show Change Note field for an unpublished document" do
+      assert page.has_no_text?("Add a public change note")
+      assert page.has_no_text?("Telling users when published information has changed is important for transparency.")
+      assert page.has_no_field?("edition[change_note]")
+    end
+
+    should "update and show success message" do
+      fill_in "edition[title]", with: "Changed Title"
+      fill_in "edition[overview]", with: "Changed Meta tag description"
+      choose("Yes")
+      fill_in "edition[body]", with: "Changed body"
+      click_button("Save")
+
+      assert page.has_field?("edition[title]", with: "Changed Title")
+      assert page.has_field?("edition[overview]", with: "Changed Meta tag description")
+      assert find(".gem-c-radio input[value='1']").checked?
+      assert page.has_field?("edition[body]", with: "Changed body")
+      assert page.has_text?("Edition updated successfully.")
+    end
+  end
+
+  context "edit tab" do
```

Comment:
We should probably avoid having two contexts with the same name ("edit tab")…

Maybe we should add some sub-contexts so we have something like:
```ruby
  context "edit tab" do
    context "editing a draft edition of a new publication" do
      setup do
        visit_edition_in_draft
        click_link("Edit") # Do we need this? We don't have it for the previously-published test
      end

      …tests
    end

    context "editing a draft edition of a previously-published publication" do
    end
  end
```

With those extra contexts, we probably then don't need to duplicate that context in the test names.

Reply:
> We should probably avoid having two contexts with the same name ("edit tab")…
> 
> Maybe we should add some sub-contexts so we have something like:
> 
> ```ruby
>   context "edit tab" do
>     context "editing a draft edition of a new publication" do
>       setup do
>         visit_edition_in_draft
>         click_link("Edit") # Do we need this? We don't have it for the previously-published test
>       end
> 
>       …tests
>     end
> 
>     context "editing a draft edition of a previously-published publication" do
>     end
>   end
> ```
> 
> With those extra contexts, we probably then don't need to duplicate that context in the test names.

Oh yes, didn't even notice we had that duplication of context ... I've added the two sub-contexts to the main one. (I'll squash this commit if it looks good.)